tests/test_container_provider.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+from mock import Mock
+
+from ansible.module_utils.basic import AnsibleModule
+
+from miqclient.api import API
+import manageiq
+
+
+@pytest.fixture(autouse=True)
+def miq_api_class(monkeypatch):
+    miq_api_class = Mock(spec=API)
+    monkeypatch.setattr("manageiq.MiqApi", miq_api_class)
+    yield miq_api_class
+
+
+@pytest.fixture
+def the_provider():
+    the_provider = Mock()
+    the_provider.name = "The provider"
+    yield the_provider
+
+
+@pytest.fixture
+def miq_ansible_module():
+    miq_ansible_module = Mock(spec=AnsibleModule)
+    yield miq_ansible_module
+
+
+@pytest.fixture
+def endpoints(miq):
+    endpoints = miq.generate_endpoints("The hostname", "12345", "The token",
+                                       "The HHostname", "54321")
+    yield endpoints
+
+
+class AnsibleModuleFailed(Exception):
+    pass
+
+
+@pytest.fixture()
+def miq(miq_api_class, miq_ansible_module, the_provider):
+    miq_ansible_module.params = {'metrics': True}
```

Comment:
This would ideally be inside test cases that need it.
But currently IIUC they all need it, as `generate_endpoints()` peeks inside `self.module.params`, which should be refactored.